src/apps/inventory/views.py

```diff
@@ -35,18 +35,53 @@ def get_context_data(self, **kwargs):
 class ProductBySellerListView(LoginRequiredMixin, ListView):
     model = Product
     paginate_by = 10
-    template_name = "product_list.html"
+    template_name = "product_by_seller_list.html"
 
     def get_queryset(self):
         if self.request.user.is_authenticated:
             return Product.objects.filter(seller=self.request.user)
         return Product.objects.none()
 
 
+class ProductFilter(FilterSet):
+    name = CharFilter(field_name="name", lookup_expr="icontains", label="Name contains")
+    category = ModelChoiceFilter(
+        field_name="category", queryset=Category.objects.all(), label="Category"
+    )
+
+    class Meta:
+        model = Product
+        fields = {
+            "price": ["lt", "gt"],
+        }
+
+
 class ProductListView(ListView):
     model = Product
-    paginate_by = 10
     template_name = "product_list.html"
+    context_object_name = "products"
+    filterset_class = ProductFilter
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+
+        if self.request.GET:
+            queryset = self.filterset_class(self.request.GET, queryset=queryset).qs
+
+            order_by = self.request.GET.get("order_by")
+            if order_by:
+                allowed_ordering_fields = ["name", "-name", "price", "-price"]
+                if order_by in allowed_ordering_fields:
+                    queryset = queryset.order_by(order_by)
+                else:
+                    queryset = queryset.order_by("-price")
+
+        return queryset
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["filter"] = self.filterset_class(self.request.GET)
```

Comment:
we create filter object on line 69
maybe we can make use of it here as well?

Reply:
Yes and i am using it there. The only option that i could think of is to run filter_class(self.request.get) before the methods and pass it but i cant do that due to not having request there, and i am not sure how to pass it otherwise